diff --git a/config.def.h b/config.def.h
index 6e0816f..d02336d 100644
--- a/config.def.h
+++ b/config.def.h
@@ -2,22 +2,22 @@
 
 /* appearance */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
-static const unsigned int gappx     = 16;       /* border gap pixel of windows */
-static const unsigned int snap      = 32;       /* snap pixel */
+static const unsigned int gappx     = 8;        /* border gap pixel of windows */
+static const unsigned int snap      = 16;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
+static const char *fonts[]          = { "Inconsolata:size=9:antialias=true" };
+static const char dmenufont[]       = "Inconsolata:size=9:antialias=true";
 
 static const XRenderColor *colors[][3] = {
 	/*               fg                     bg                 border               */
-	[SchemeNorm] = { RGBA(bb,bb,bb,OPAQUE), RGBA(22,22,22,d0), RGBA(44,44,44,OPAQUE) },
-	[SchemeSel] =  { RGBA(ee,ee,ee,OPAQUE), RGBA(00,55,77,d0), RGBA(00,55,77,OPAQUE) },
-	[SchemeUrg] =  { RGBA(ee,ee,ee,OPAQUE), RGBA(ee,44,44,d0), RGBA(ee,44,44,OPAQUE) },
+	[SchemeNorm] = { RGBA(60,60,60,OPAQUE), RGBA(20,20,20,d0), RGBA(69,69,69,OPAQUE) },
+	[SchemeSel] =  { RGBA(aa,aa,aa,OPAQUE), RGBA(20,20,20,d0), RGBA(69,69,69,OPAQUE) },
+	[SchemeUrg] =  { RGBA(ce,5c,00,OPAQUE), RGBA(20,20,20,d0), RGBA(69,69,69,OPAQUE) },
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "I", "II", "III", "IV", "V" };
 
 static const Rule rules[] = {
 	/* xprop(1):
@@ -54,7 +54,7 @@ static const Layout layouts[] = {
 
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
-static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", "#222", "-nf", "#bbb", "-sb", "#057", "-sf", "#eee", NULL };
+static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", "#202020", "-nf", "#606060", "-sb", "#202020", "-sf", "#aaaaaa", NULL };
 static const char *termcmd[]  = { "st", NULL };
 
 static Key keys[] = {
